src/Sankey.js

```diff
@@ -0,0 +1,287 @@
+/**
+    @external Viz
+    @see https://github.com/d3plus/d3plus-viz#Viz
+*/
+import {nest} from "d3-collection";
+import {sankey, sankeyLinkHorizontal} from "d3-sankey";
+
+import {accessor, assign, configPrep, constant, elem} from "d3plus-common";
+import {Path} from "d3plus-shape";
+import * as shapes from "d3plus-shape";
+import {dataLoad as load, Viz} from "d3plus-viz";
+
+/**
+    @class Sankey
+    @extends external:Viz
+    @desc Creates a sankey visualization based on a defined set of nodes and links. [Click here](http://d3plus.org/examples/d3plus-network/sankey-diagram/) for help getting started using the Sankey class.
+*/
+export default class Sankey extends Viz {
+
+  /**
+      @memberof Sankey
+      @desc Invoked when creating a new class instance, and sets any default parameters.
+      @private
+  */
+  constructor() {
+    super();
+    this._nodeId = accessor("id");
+    this._links = accessor("links");
+    this._noDataMessage = false;
+    this._nodes = accessor("nodes");
+    this._nodeWidth = 30;
+    this._on.mouseenter = () => {};
+    this._on["mouseleave.shape"] = () => {
+      this.hover(false);
+    };
+    this._on["mousemove.shape"] = (d, i) => {
+      if (this._focus && this._focus === d.id) {
+        this.hover(false);
+        this._on.mouseenter.bind(this)(d, i);
+
+        this._focus = undefined;
+      }
+      else {
+        const id = this._nodeId(d, i),
+              node = this._nodeLookup[id],
+              nodeLookup = Object.keys(this._nodeLookup).reduce((all, item) => {
+                all[this._nodeLookup[item]] = !isNaN(item) ? parseInt(item) : item;
+                return all;
+              }, {});
+
+        const links = this._linkLookup[node];
+        const filterIds = [id];
+
+        links.forEach(l => {
+          filterIds.push(nodeLookup[l]);
+        });
+
+        this.hover((h, x) => {
+          if (h.source && h.target) {
+            return h.source.id === id || h.target.id === id;
+          }
+          else {
+            return filterIds.includes(this._nodeId(h, x));
+          }
+        });
+      }
+    };
+    this._path = sankeyLinkHorizontal();
+    this._sankey = sankey();
+    this._shape = constant("Rect");
+    this._shapeConfig = assign(this._shapeConfig, {
+      Path: {
+        fill: "none",
+        hoverStyle: {
+          "stroke-width": d =>
+            Math.max(
+              1,
+              Math.abs(d.source.y1 - d.source.y0) * (d.value / d.source.value) -
+                15
+            )
+        },
+        label: false,
+        stroke: "#DBDBDB",
+        strokeOpacity: 0.2,
+        strokeWidth: d =>
+          Math.max(
+            1,
+            Math.abs(d.source.y1 - d.source.y0) * (d.value / d.source.value) -
```

Comment:
we cannot reference `.value` here directly, please use the `this._value` accessor function